### PR TITLE
html5 google form

### PR DESCRIPTION
**Pull request by krash1408**
### html5 google form

[Demo](https://krash1408.github.io/google-header/) | [Code base](https://github.com/krash1408/frontend-2019-homeworks/tree/googlelike-form/submissions/krash1408/html5form)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/krash1408/html5form/style.css [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+  padding: 0;
+}
+body {
+  font-size: 1.3rem;
+  font-family: sans-serif;
+  margin: 0;
+  padding: 0;
+}
+.visually-hidden {
+  position: absolute !important;
```
Comment (A-Ostrovnyy) [inline submissions/krash1408/html5form/style.css]: Avoid using `!important` rules.

File: submissions/krash1408/html5form/style.css [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+  clip: rect(1px 1px 1px 1px); /* IE6, IE7 */
+  clip: rect(1px, 1px, 1px, 1px);
```
Comment (A-Ostrovnyy) [inline submissions/krash1408/html5form/style.css]: Better use `clip: rect(0 0 0 0)`.

File: submissions/krash1408/html5form/style.css [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+
+/* e-mail & pictures links */
+.main-link {
+  text-decoration: none;
+  color: #777;
+  margin: 0.5rem;
+  font-size: 1.4rem;
+  cursor: pointer;
+  will-change: color;
+  transition: color 0.3s;
```
Comment (A-Ostrovnyy) [inline submissions/krash1408/html5form/style.css]: A good practice is adding a timing function to `transition` property.

File: submissions/krash1408/html5form/style.css [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+.main-btn__svg {
+  width: 100%;
+  height: 100%;
+}
+.main-btn__svg path {
+  fill: #777;
+  will-change: fill;
+  transition: fill .3s;
+}
+#menu:focus + .main-btn > .main-btn__svg path {
```
Comment (A-Ostrovnyy) [inline submissions/krash1408/html5form/style.css]: Avoid styling by `#id`.

File: submissions/krash1408/html5form/index.html [hunk 5]
```diff
@@ -0,0 +1,402 @@
...
+      </label>
+
+      <div class="dropdown-menu">
+        <input
+          class="visually-hidden dropdown-menu__btn"
+          type="checkbox"
+          name=""
+          id="dropdown-menu__btn"
+        />
+        <label class="btn__label" for="dropdown-menu__btn">Ещё</label>
```
Comment (A-Ostrovnyy) [inline submissions/krash1408/html5form/index.html]: Button More not accessible from the keyboard.

File: submissions/krash1408/html5form/style.css (hunk 2 above)
Comment (krash1408) [reply submissions/krash1408/html5form/style.css]: I took this class from [HERE](https://gist.github.com/marcol/3979586). HtmlAcademy advise to use it...

File: submissions/krash1408/html5form/style.css (hunk 3 above)
Comment (krash1408) [reply submissions/krash1408/html5form/style.css]: Ok. Will fix it.

File: submissions/krash1408/html5form/style.css (hunk 1 above)
Comment (krash1408) [reply submissions/krash1408/html5form/style.css]: I took this class from [HERE](https://gist.github.com/marcol/3979586). HtmlAcademy advise to use it...

File: submissions/krash1408/html5form/style.css (hunk 4 above)
Comment (krash1408) [reply submissions/krash1408/html5form/style.css]: Okey. Will fix soon.

File: submissions/krash1408/html5form/index.html (hunk 5 above)
Comment (krash1408) [reply submissions/krash1408/html5form/index.html]: Button is accessible. after you go through all the main menu items (notification, profile).

Comment (krash1408) [reply submissions/krash1408/html5form/index.html]: I dont understand, how to fix an issue with order in this case.

File: submissions/krash1408/html5form/style.css (hunk 2 above)
Comment (A-Ostrovnyy) [reply submissions/krash1408/html5form/style.css]: LOL, Vadim Makeev in the one of last his video recommended using `clip: rect(0 0 0 0)`.

File: submissions/krash1408/html5form/index.html (hunk 5 above)
Comment (A-Ostrovnyy) [reply submissions/krash1408/html5form/index.html]: First of all, you iterating the menu items. If you open a popup after pressing Tab key your focus must go to the popup's items, after iterating first part popups items focus must go to the button More. Opening the second part of the popup must be available from the keyboard, and after pressing Tab key focus must go to the first element of the second part of the popup.

File: submissions/krash1408/html5form/style.css (hunk 2 above)
Comment (krash1408) [reply submissions/krash1408/html5form/style.css]: Don't see it. Ok. I'll fix it.

File: submissions/krash1408/html5form/index.html (hunk 5 above)
Comment (krash1408) [reply submissions/krash1408/html5form/index.html]: Ok. I'll try to implement it. Thanks a lot for help!

Comment (krash1408) [reply submissions/krash1408/html5form/index.html]: I guess, it now works fine with the exception of main menu items. I don't now, how to skip them off from  focusing when the menu is active...

File: submissions/krash1408/html5form/style.css (hunk 2 above)
Comment (krash1408) [reply submissions/krash1408/html5form/style.css]: fixed.

File: submissions/krash1408/html5form/style.css (hunk 1 above)
Comment (krash1408) [reply submissions/krash1408/html5form/style.css]: This class may always working in priority. And IMHO, in this situation '!importante' is usefull case.

File: submissions/krash1408/html5form/index.html (hunk 5 above)
Comment (A-Ostrovnyy) [reply submissions/krash1408/html5form/index.html]: Order of iteration element depends on order these elements in your markup(`html`).

File: submissions/krash1408/html5form/style.css (hunk 1 above)
Comment (krash1408) [reply submissions/krash1408/html5form/style.css]: Ok. I finnaly delete it.

File: submissions/krash1408/html5form/index.html (hunk 5 above)
Comment (krash1408) [reply submissions/krash1408/html5form/index.html]: IMHO, now all clear ;)

**Comment by A-Ostrovnyy** [comment]
Well done!

**Pull request merged**